tests/functional/cylc-clean/00-basic.t

```diff
@@ -33,6 +33,7 @@ create_test_global_config "" "
         [[[localhost]]]
             run = ${TEST_DIR}/${SYM_NAME}/run
             log = ${TEST_DIR}/${SYM_NAME}/log
+            log/job = ${TEST_DIR}/${SYM_NAME}/log
```

Comment:
I tried to break the clean functionality by doing this:

```diff
diff --git a/cylc/flow/clean.py b/cylc/flow/clean.py
index aec2c4efe..acd49b8ab 100644
--- a/cylc/flow/clean.py
+++ b/cylc/flow/clean.py
@@ -212,6 +212,12 @@ def clean(id_: str, run_dir: Path, rm_dirs: Optional[Set[str]] = None) -> None:
 
     """
     symlink_dirs = get_symlink_dirs(id_, run_dir)
+    symlink_dirs = {
+        key: value
+        for key, value in symlink_dirs.items()
+        if key != 'log/job'
+    }
+
     if rm_dirs is not None:
         # Targeted clean
         for pattern in rm_dirs:
```

The test passed, which suggests that it isn't actually testing that the `log/job` dir gets removed, only that it is being created?

My methodology could be wrong, or this could be covered by another test. Please check that this is being tested.

Reply:
This test doesn't pass for me with this change - 

```
+/var/tmp/tmp.Gm3l7UNspz/FDEk2IkM3G/job/cylc-run/cylctb-20240710T154301Z-UgZU
+`-- functional
+    `-- cylc-clean
+        `-- 01-remote
+            `-- log
+                `-- job
+                    `-- 1
+                        `-- santa
+                            |-- 01
+                            |   |-- job
+                            |   |-- job.err
+                            |   |-- job.out
+                            |   `-- job.status
+                            `-- NN -> 01
not ok 10 - test-dir-tree-post-clean.remote.stdout-cmp-ok
```

Which pretty much exactly what I'd expect.

Reply:
Yep, I can confirm that the remote test `01-remote` does fail, however, the local test `00-basic` (which was changed to add a `log/job` symlink) does not.

Reply:
I can't replicate the problem with 00-basic!

Reply:
```console
$ git show --oneline
babe65701 (HEAD, tim/feat.symlink_log.job) Update cylc/flow/workflow_files.py
diff --git a/cylc/flow/workflow_files.py b/cylc/flow/workflow_files.py
index c2655036e..c20558d79 100644
--- a/cylc/flow/workflow_files.py
+++ b/cylc/flow/workflow_files.py
@@ -264,7 +264,7 @@ class WorkflowFiles:
     RESERVED_NAMES = frozenset([FLOW_FILE, SUITE_RC, *RESERVED_DIRNAMES])
     """Reserved filenames that cannot be used as run names."""
 
-    SYMLINK_DIRS = frozenset(list(SYMLINKABLE_LOCATIONS.keys()) + [''])
+    SYMLINK_DIRS = frozenset(list(SYMLINKABLE_LOCATIONS) + [''])
     """The paths of the symlink dirs that may be set in
     global.cylc[install][symlink dirs], relative to the run dir
     ('' represents the run dir)."""
$ git diff
diff --git a/cylc/flow/clean.py b/cylc/flow/clean.py
index aec2c4efe..acd49b8ab 100644
--- a/cylc/flow/clean.py
+++ b/cylc/flow/clean.py
@@ -212,6 +212,12 @@ def clean(id_: str, run_dir: Path, rm_dirs: Optional[Set[str]] = None) -> None:
 
     """
     symlink_dirs = get_symlink_dirs(id_, run_dir)
+    symlink_dirs = {
+        key: value
+        for key, value in symlink_dirs.items()
+        if key != 'log/job'
+    }
+
     if rm_dirs is not None:
         # Targeted clean
         for pattern in rm_dirs:
$ ctb -v tests/functional/cylc-clean/00-basic.t

ok 1 - 00-basic-val
ok 2 - run-dir-readlink-pre-clean.stdout-cmp-ok
ok 3 - test-dir-tree-pre-clean
ok 4 - test-dir-tree-pre-clean.stdout-cmp-ok
ok 5 - clean
ok 6 - 00-basic-file-exists-fail
ok 7 - test-dir-tree-post-clean
ok 8 - test-dir-tree-post-clean.stdout-cmp-ok
ok 9 - clean-non-exist
ok 10 - clean-non-exist.stdout-cmp-ok
ok    13788 ms
[13:20:55]
All tests successful.
Files=1, Tests=10, 22 wallclock secs ( 0.04 usr  0.02 sys +  7.35 cusr  2.22 csys =  9.63 CPU)
Result: PASS
```

Reply:
Sorry - I've just re-read this whole thing and I realized that I've confused myself. You've broken my change and expect this test to fail, but it doesn't. 

Willfix